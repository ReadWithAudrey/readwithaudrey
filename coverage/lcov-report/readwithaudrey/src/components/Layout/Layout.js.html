<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/components/Layout/Layout.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/components/Layout</a> Layout.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/11</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/*eslint-disable*/
import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'
&nbsp;
import 'normalize.css'
import 'tachyons'
import '../../styles/index.css'
&nbsp;
import { Header, Footer } from '../index.js'
&nbsp;
export const PureLayout = ({ children, data }) =&gt; {
  const title = data !== undefined ? data.site.siteMetadata.title : 'Audrey'
  return (
    &lt;React.Fragment&gt;
      &lt;Helmet
        title={title}
        meta={[
          { name: 'description', content: 'Sample' },
          { name: 'keywords', content: 'sample, something' },
        ]}
      &gt;
        &lt;html lang="en" /&gt;
      &lt;/Helmet&gt;
      &lt;Header siteTitle={title} /&gt;
      &lt;div className="flex flex-column items-center justify-center page-padding ph4"&gt;
        {children}
      &lt;/div&gt;
      &lt;Footer /&gt;
    &lt;/React.Fragment&gt;
  )
}
&nbsp;
const QueryLayout = <span class="fstat-no" title="function not covered" >({</span> children }) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;StaticQuery</span>
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={<span class="fstat-no" title="function not covered" >da</span>ta =&gt; <span class="cstat-no" title="statement not covered" >&lt;PureLayout data={data}&gt;{children}&lt;/PureLayout&gt;}</span>
  /&gt;
)
&nbsp;
PureLayout.propTypes = {
  children: PropTypes.node,
  data: PropTypes.object,
}
QueryLayout.propTypes = {
  children: PropTypes.node,
}
&nbsp;
let Layout
<span class="missing-if-branch" title="else path not taken" >E</span>if (process.env.NODE_ENV === 'test') {
  Layout = PureLayout
} else {
<span class="cstat-no" title="statement not covered" >  Layout = QueryLayout</span>
}
&nbsp;
export default Layout
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
