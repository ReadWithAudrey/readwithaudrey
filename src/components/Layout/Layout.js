/*eslint-disable*/
import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'

import { Header, Footer } from '../index.js'

export const PureLayout = ({ children, data }) => {
  console.log(typeof data);
  return(
  <React.Fragment>
    <Helmet
      title={data.site.siteMetadata.title}
      meta={[
        { name: 'description', content: 'Sample' },
        { name: 'keywords', content: 'sample, something' },
      ]}
    >
      <html lang="en" />
    </Helmet>
    <Header siteTitle={data.site.siteMetadata.title} />
    {children}
    <Footer />
  </React.Fragment>
)}

export const Layout = ({ children }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => <PureLayout data={data}>{children}</PureLayout>}
  />
)

PureLayout.propTypes = {
  children: PropTypes.node,
  data: PropTypes.object
}
Layout.propTypes = {
  children: PropTypes.node,
}

export default Layout
