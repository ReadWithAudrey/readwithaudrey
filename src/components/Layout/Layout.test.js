import React from 'react'
import renderer from 'react-test-renderer'
import { PureLayout as Layout } from './Layout'

describe('Layout component snapshot test', () =>
  it('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'Audrey',
        },
      },
    }
    const tree = renderer.create(<Layout data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
