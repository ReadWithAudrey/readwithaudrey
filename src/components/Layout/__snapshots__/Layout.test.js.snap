// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component snapshot test renders correctly 1`] = `
Array [
  <header
    className="sc-bdVaJa flex items-center justify-between ph3 pv2 montserrat white vw-100 f3 z-max bqQFjy"
  >
    <div
      className="flex items-center"
      id="header-left"
    >
      <Link
        to="/"
      >
        <img
          alt="book icon"
          className="h2 ph2"
          src="test-file-stub"
        />
      </Link>
      <Link
        to="/"
      >
        Audrey
      </Link>
    </div>
    <nav
      className="sc-bwzfXH w-65 w-40-l montserrat f4 gQAWUv"
    >
      <section
        className="sc-htpNat montserrat jQdasv"
      >
        <Link
          to="/why/"
        >
          Why?
        </Link>
        <Link
          to="/faq/"
        >
          Questions
        </Link>
        <Link
          to="/contact/"
        >
          Contact
        </Link>
        <a
          className="sc-EHOje white br2 ba bw1 b--white jbOSZe"
          href="/Form1"
        >
          Get Started
        </a>
      </section>
      <section
        className="sc-bxivhb igYaY"
      >
        <a
          className="dtc fr tr v-mid dn-ns p0-ns link dim white"
          onClick={[Function]}
        >
          <span
            className="db w2 mt1 mb2 bw2 pb2 bt bb b--white"
          />
          <span
            className="db w2 mt1 bw2 pb1 bt b--white"
          />
        </a>
      </section>
    </nav>
  </header>,
  <div
    className="page-padding ph4 ph7-ns vw-100 center mw9"
  />,
  <footer
    className="sc-bZQynM flex flex-column justify-center items-center montserrat w-100 bt bw2 mid-gray cYxVbM"
  >
    <span
      className="sc-gzVnrw flex justify-center items-center bg-white mid-gray w-100 pa2 hZtFAF"
    >
      <p
        className="mid-gray ph2 ph4-ns f5"
      >
        Â© Audrey
      </p>
      <Link
        className="no-underline mid-gray ph2 ph4-ns"
        to="/terms/"
      >
        Terms
      </Link>
      <Link
        className="no-underline mid-gray ph2 ph4-ns"
        to="/privacy/"
      >
        Privacy
      </Link>
      <Link
        className="no-underline mid-gray ph2 ph4-ns"
        to="/contact/"
      >
        Contact
      </Link>
    </span>
  </footer>,
]
`;
